backend/s3/file.go

```diff
@@ -66,13 +56,10 @@ func (f *File) Path() string {
 // the object's HEAD through the s3 API.
 func (f *File) Exists() (bool, error) {
 	_, err := f.getHeadObject()
-	code := ""
 	if err != nil {
-		code = err.(awserr.Error).Code()
-	}
-	if err != nil && (code == s3.ErrCodeNoSuchKey || code == "NotFound") {
-		return false, nil
-	} else if err != nil {
+		if errors.Is(err, vfs.ErrNotExist) {
+			return false, nil
+		}
 		return false, err
 	}
 
```

Comment:
this logic is largely moved to handleExistsError which is called directly in the getHeadObject call, so now we just look for the sentinel error, vfs.ErrNotExist.